### ISSUE TITLE
Update User Schema Resources and add User Type

### ISSUE THREAD

**Issue by RichiCoder1** [okta/terraform-provider-okta#21]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It looks like the User Schema resources as defined here have fallen behind how they're currently implemented. Namely Schemas correspond to a User Type (which also doesn't appear to be supported.

### New or Affected Resource(s)

* okta_app_user_base_schema
* okta_app_user_schema
* okta_user_base_schema
* okta_user_schema
* okta_user_type (new)

### Potential Terraform Configuration

```hcl
resource "okta_user_type" "contractor" {
  description    = "Third Party Contractor"
  displayName    = "Contractor"
  # Should name be pulled implictly?
  # name         = "contractor"
}

```

### References

* https://developer.okta.com/docs/reference/api/user-types/
* https://developer.okta.com/docs/reference/api/schemas/

**Comment by RichiCoder1** [comment]
*bump* not stale, still an issue

**Comment by rajnadimpalli** [comment]
issue is not stale

**Comment by bpottier** [comment]
Is there any update on this?

**Comment by rajnadimpalli** [comment]
Hi @noinarisak ,

I have submitted a pull request for this enhancement (https://github.com/oktadeveloper/terraform-provider-okta/pull/183).
Can you please help to review?

Thank you.

**Comment by bogdanprodan-okta** [comment]
Hi @RichiCoder1 @bpottier @rajnadimpalli! We are aware that terraform is missing this feature and it will be released in the new v3.7.0 version. I will review #183 shortly and release a new version soon.

**Comment by bogdanprodan-okta** [comment]
Hi @RichiCoder1 @bpottier @rajnadimpalli! The issue should be fixed in the latest 3.7.2 version. In case it still exists, please reopen this issue.
